nixos/modules/services/network-filesystems/seaweedfs.nix

```diff
@@ -0,0 +1,597 @@
+      };
+
+      readMode = lib.mkOption {
+        type = lib.types.enum [
+          "local"
+          "proxy"
+          "redirect"
+        ];
+        default = "proxy";
+        description = "How to deal with non-local volume: not found|proxy to remote node|redirect volume location.";
+      };
+
+      minFreeSpace = lib.mkOption {
+        type = lib.types.str;
+        default = "1";
+        description = "Min free disk space (value<=100 as percentage like 1, other as human readable bytes, like 10GiB).";
+      };
+
+      fileSizeLimitMB = lib.mkOption {
+        type = lib.types.int;
+        default = 256;
+        description = "Limit file size to avoid out of memory.";
+      };
+
+      metricsPort = lib.mkOption {
+        type = lib.types.nullOr lib.types.port;
+        default = null;
+        description = "Prometheus metrics listen port.";
+      };
+
+      whiteList = lib.mkOption {
+        type = lib.types.listOf lib.types.str;
+        default = [ ];
+        description = "Ip addresses having write permission. No limit if empty.";
+      };
+    };
+
+    filer = {
+      enable = lib.mkEnableOption "SeaweedFS filer server";
+
+      port = lib.mkOption {
+        type = lib.types.port;
+        default = 8888;
+        description = "Port for filer server.";
+      };
+
+      grpcPort = lib.mkOption {
+        type = lib.types.port;
+        default = 18888;
+        description = "gRPC port for filer server.";
+      };
+
+      ip = lib.mkOption {
+        type = lib.types.str;
+        description = "IP address to bind to.";
+      };
+
+      ipBind = lib.mkOption {
+        type = lib.types.str;
+        default = "";
+        description = "IP address to bind to. If empty, defaults to same as ip.";
+      };
+
+      collection = lib.mkOption {
+        type = lib.types.str;
+        default = "";
+        description = "All data will be stored in this default collection.";
+      };
+
+      dataDir = lib.mkOption {
+        type = lib.types.str;
+        default = "${baseDir}/filer";
+        description = "Data directory for filer.";
+      };
+
+      defaultReplicaPlacement = lib.mkOption {
+        type = lib.types.str;
+        default = "";
+        description = "Default replication type. If not specified, use master setting.";
+      };
+
+      master = lib.mkOption {
+        type = lib.types.listOf lib.types.str;
+        default = [ ];
+        example = [
+          "192.168.1.10:9333"
+          "192.168.1.11:9333"
+        ];
+        description = "List of master servers addresses.";
+      };
+
+      maxMB = lib.mkOption {
+        type = lib.types.int;
+        default = 4;
+        description = "Split files larger than the limit.";
+      };
+
+      metricsPort = lib.mkOption {
+        type = lib.types.nullOr lib.types.port;
+        default = null;
+        description = "Prometheus metrics listen port.";
+      };
+
+      tomlConfig = lib.mkOption {
```

Comment:
```suggestion
      config = lib.mkOption {
```

Reply:
Hi,

You must understand that I deliberately did not put config because it leads to confusion or interpretation because this file is not at all necessary, only optional for certain use cases.
I don't really like the idea of ​​renaming it to config I prefer because I wouldn't want people to be misled about it.

Best Regards